pkg/scripts/os_centos.go

```diff
@@ -52,6 +52,13 @@ gpgcheck=1
 repo_gpgcheck=0
 gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
 EOF
+
+os_name=$(cat /etc/os-release | head -n1 | cut -d \" -f2)
+os_version=$(cat /etc/os-release | head -n2 | tail -1 | cut -d \" -f2)
+if [ "$os_name" == "CentOS Linux" ] && [ "$os_version" == "8 (Core)" ]; then
+	sudo sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+	sudo sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
+fi
```

Comment:
I have mixed fillings about this:
* it has to be ported to MC and OSM
* this is band-aid
* it's prologues the agony of already dead centos8 :zombie: 

What would help is if kubeone learns to recognizing other RHEL rebuilds and act on them as they were "just centos". There are two non-contradicting ways to do this:
* open `HostConfig.OperatingSystem` to be set from the config/terraform by the user (like we did with Hostname)
* if `HostConfig.OperatingSystem` only then use automatic detection, and at this step we need to teach K1 to recognize all those RockyLinux/OracleLinux/AlmaLinux and other RHEL rebuilds.

Reply:
Since we already set OS anyway in the machineDeployments, opening `HostConfig.OperatingSystem` to be set from the config seems like the easiest way to help those in transition.